src/msg/async/AsyncConnection.cc

```diff
@@ -821,6 +816,8 @@ void AsyncConnection::process()
     }
   } while (prev_state != state);
 
+  if (need_dispatch_writer)
```

Comment:
I see - I assumed a check in handle_write() was sufficient, but it's not.  is_connected() can be used to make it slightly more clear, right?